pkgs/by-name/fa/factorio/utils.nix

```diff
@@ -83,4 +84,45 @@ in
         cp $src $out/$srcBase
       '';
     };
+
+  modsFromLock =
+    lockFile:
+    let
+      lock = builtins.fromJSON (builtins.readFile lockFile);
+      downloadMod =
+        {
+          name,
+          version,
+          download_url,
+          file_name,
+          sha1,
+        }:
+        let
+          # It is up to the user to tell curl how to authenticate against mods.factorio.com.
+          # The easiest way is to prefetch the file.
+          src = fetchurl {
+            name = file_name;
+            url = "https://mods.factorio.com${download_url}";
+            inherit sha1;
+          };
```

Comment:
IIRC you're supposed to use "requireFile", my repo does provide a method to automate downloads tho, it's a bit hacky.

Reply:
I agree that you don't want to include the secrets in nix store. There exists .netrc that Nix somewhat supports, but I don't think Factorio's API supports basic auth. So even that is out of the question and we should just use `requireFile` here.